rclcpp/test/benchmark/benchmark_node.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+class NodePerformanceTest : public PerformanceTest
+{
+public:
+  void SetUp(::benchmark::State & state)
+  {
+    rclcpp::init(0, nullptr);
+    performance_test_fixture::PerformanceTest::SetUp(state);
+  }
+
+  void TearDown(::benchmark::State & state)
+  {
+    rclcpp::shutdown();
+    performance_test_fixture::PerformanceTest::TearDown(state);
```

Comment:
Since `PerformanceTest::TearDown` stops the memory measurements, you should call that first so that any memory operations involved in tearing down `rclcpp` don't get recorded.

```suggestion
    performance_test_fixture::PerformanceTest::TearDown(state);
    rclcpp::shutdown();
```